### ISSUE TITLE
https_proxy support to download tuf-root from https://sigstore-tuf-root.storage.googleapis.com/

### ISSUE THREAD

**Issue by alekdu** [sigstore/policy-controller#673]
**Description**
Hello experts! :-)
I met a trouble to config the policy-controller pod to read TUF root data from outside world (yes, we are in inner network and need a proxy server to access outside). Seems whatever I set the proxy environment, the policy-controller just ignores it anyway. Any thoughts? Thanks!

**Comment by hectorj2f** [comment]
@alekdu Thanks for opening the issue. We're just using https://github.com/theupdateframework/go-tuf library to read the remote TUF root. Generally the golang network package should automatically use the HTTPS_PROXY if you set that environment variable.

**Comment by alekdu** [comment]
@hectorj2f thanks for your reply!

As I double checked, it seems I should have set the environment variables correctly:

cat /etc/systemd/system/docker.service.d/http-proxy.conf
[Service]
Environment="HTTP_PROXY=http://<myproxy>:<myport>" "HTTPS_PROXY=http://<myproxy>:<myport>" "NO_PROXY=<.....>"

Still I got the https connection error as:

{"level":"info","ts":1679558460.9429631,"logger":"fallback","caller":"webhook/main.go:94","msg":"Initializing TUF root from  => https://sigstore-tuf-root.storage.googleapis.com"}
{"level":"panic","ts":1679558610.969934,"logger":"fallback","caller":"webhook/main.go:96","msg":"Failed to initialize TUF client from  : updating local metadata and targets: error updating to TUF remote mirror: tuf: failed to download 3.root.json: Get \"https://sigstore-tuf-root.storage.googleapis.com/3.root.json\": dial tcp 142.251.39.112:443: i/o timeout\nremote status:{\n\t\"mirror\": \"https://sigstore-tuf-root.storage.googleapis.com\",\n\t\"metadata\": {}\n}","stacktrace":"main.main\n\tgithub.com/sigstore/policy-controller/cmd/webhook/main.go:96\nruntime.main\n\truntime/proc.go:250"}
panic: Failed to initialize TUF client from  : updating local metadata and targets: error updating to TUF remote mirror: tuf: failed to download 3.root.json: Get "https://sigstore-tuf-root.storage.googleapis.com/3.root.json": dial tcp 142.251.39.112:443: i/o timeout

**Comment by alekdu** [comment]
Ok, sounds like I have to change the policy-controller deployment ENV as:

env:
     - name: HTTPS_PROXY
       value: <value>
     - name: NO_PROXY
       value: <value>

Is there anyway that the go POD can take the global environment automatically ?

**Comment by hectorj2f** [comment]
@alekdu Yes, you can set the env variables using our helm chart https://github.com/sigstore/helm-charts/blob/main/charts/policy-controller/values.yaml#L18 and https://github.com/sigstore/helm-charts/blob/main/charts/policy-controller/values.yaml#L60.

**Comment by hectorj2f** [comment]
Thanks @alekdu
